src/OSPSuite.R/Services/ContainerTask.cs

```diff
@@ -115,6 +115,27 @@ public IParameter[] AllParametersForSensitivityAnalysisMatching(ISimulation simu
 
       public string[] AllStateVariableParameterPathsIn(IModelCoreSimulation simulation) => AllStateVariableParameterPathsIn(simulation?.Model?.Root);
 
+      public String[] BaseUnitNameFromPath(IModelCoreSimulation simulation, string path)
```

Comment:
WOW what is this method doing here.  Can you explain the user case?
AT any rate we need

1. Method to be defined in the interface (IContainerTask) WITH comments explaining what it does
2. Use string not String
3. Use plurals since it returns an array (BaseUnitNames)
4. You will have duplicates in this code (mg etc.. will be multiple time I guess depending on the path)
5. Or is path supposed to be a REAL path?
Anyways please explain use case and I can update

Reply:
I can address the points but probably not today.

In short - it will be used in R in getOutputValues

https://github.com/Open-Systems-Pharmacology/OSPSuite-R/blob/79bc93da81078f57584951fd4e98f0820cb691e4/R/utilities-simulation-results.R#L86

to get the unit and dimension without creating an object.

Reply:
but isn't `quantities` used for something else? The line you are pointing too is **not** creating a quantity

Reply:
I have re-written the method that it does not create quantities at all. If quantities instead of paths are passed, paths are extracted and then proceeded with.